### DISCUSSION TITLE
Manually paint area which to avoid

### DISCUSSION THREAD

**Discussion by thejpfin** [light-and-ray/sd-webui-replacer#25] [Ideas]
Hello, I have been testing this extension and it's pretty cool. Can you add (or maybe you already can do this?) some system where you can manually paint area which to inpaint or avoid inpainting. 
Let's say you want to change color of shirt, but only part of the shirt. Here's where manual inpainting before clicking run could come handy. 
Or maybe detection model accidentally picks up wrong objects, with manual inpaint tool you could stop that.

**Comment by light-and-ray** [comment]
Maybe in future, I thought about it too. But it kinda brake philosophy of fast inpainting: if you need to touch a lot of settings, it's better to use regular inpaint. If I add custom masks, I will try to find a good place for them in ui. Maybe inside accordion inside advanced options

**Comment by light-and-ray** [comment]
Implemented!

![Screenshot_20240202_030213](https://github.com/light-and-ray/sd-webui-replacer/assets/33491867/15ce2525-5ceb-4e0c-9abe-2af56b14ffe5)
